src/functions/indPolyhedralQPDAS.jl

```diff
@@ -0,0 +1,146 @@
+# IndPolyhedral: QPDAS implementation
+
+import QPDAS
+
+"""
+**Indicator of a polyhedral set**
+
+    IndPolyhedralQPDAS(A, C, b, d)
+
+```math
+S = \\{x : \\langle A, x \\rangle = b\\ ∧ \\langle C, x \\rangle \\le b\\}.
+```
+"""
+struct IndPolyhedralQPDAS{R<:Real, MT<:AbstractMatrix{R}, VT<:AbstractVector{R}, QP<:QPDAS.QuadraticProgram} <: IndPolyhedral
+    A::MT
+    b::VT
+    C::MT
+    d::VT
+    z::VT
+    qp::QP
+    first_prox::Ref{Bool}
+    function IndPolyhedralQPDAS{R}(A::MT, b::VT, C::MT, d::VT) where {R<:Real, MT<:AbstractMatrix{R}, VT<:AbstractVector{R}, QP<:QPDAS.QuadraticProgram}
+        @assert size(A,1) == size(b,1)
+        qp = QPDAS.QuadraticProgram(A, b, C, d, smartstart=false)
+        new{R, MT, VT, typeof(qp)}(A, b, C, d, zeros(R,size(A,2)), qp, Ref(true))
+    end
+end
+
+# properties
+
+is_prox_accurate(::IndPolyhedralQPDAS) = true
+
+# constructors
+
+function IndPolyhedralQPDAS(
+    l::AbstractVector{R}, A::AbstractMatrix{R}, u::AbstractVector{R}
+) where R
+    if !all(l .<= u)
+        error("function is improper (are some bounds inverted?)")
+    end
+    eqinds = (l .== u) .& .!isnothing.(l)
+    Aeq = A[eqinds,:]
+    beq = l[eqinds]
+
+    _islower(l::T) where T =
+        l != typemin(T) && !isnan(l) && !isnothing(l)
+    _isupper(u::T) where T =
+        u != typemax(T) && !isnan(u) && !isnothing(u)
+
+    lower = _islower.(l) .& (.!eqinds)
+    upper = _isupper.(u) .& (.!eqinds)
+
+    lower_only = lower .& (.! upper)
+    upper_only = upper .& (.! lower)
+    upper_and_lower = upper .& lower
+
+
+    Cieq = [-A[lower_only, :];
+             A[upper_only, :];
+            -A[upper_and_lower, :];
+             A[upper_and_lower, :] ]
+    dieq = [-l[lower_only];
+             u[upper_only];
+            -l[upper_and_lower];
+             u[upper_and_lower] ]
+
+    IndPolyhedralQPDAS{R}(Aeq, beq, Cieq, dieq)
+end
```

Comment:
This function is quite nasty, but I don't see another way to support the old syntax otherwise.

Reply:
It’s okay, I don’t see many other way of doing it.

I’m open to changing the overall syntax for the construction of `IndPolyhedral` objects. But maybe we can open a dedicated issue for that and focus here in meeting the current construction signature.